core/src/main/java/org/apache/iceberg/rest/responses/GetNamespaceResponse.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.ImmutablesUtil;
+
+/**
+ * Represents a REST response to fetch namespace metadata.
+ * <p>
+ * The properties returned should include all the user provided properties from the
+ * request, as well as any server-side added properties.
+ */
+public class GetNamespaceResponse {
+
+  private Namespace namespace;
+  private Map<String, String> properties;
+
+  public GetNamespaceResponse() {
+    // Required for Jackson deserialization
+  }
+
+  private GetNamespaceResponse(Namespace namespace, Map<String, String> properties) {
+    this.namespace = namespace;
+    this.properties = properties;
+    validate();
+  }
+
+  GetNamespaceResponse validate() {
+    ValidationException.check(namespace != null && !namespace.isEmpty(),
+        "Cannot build a response to get a namespace without specifying the namespace");
+    return this;
+  }
+
+  public Namespace namespace() {
+    return namespace;
+  }
+
+  public Map<String, String> properties() {
+    return ImmutablesUtil.nullSafeCopyOf(properties);
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Ah, I thought you said that this was updated, but it looks like it hasn't been so I'll stop leaving comments on it.

Reply:
No worries. It's updated now!